catalog/registry/catalog-registry-identificationplugin/src/main/java/ddf/catalog/registry/identification/IdentificationPlugin.java

```diff
@@ -76,21 +102,50 @@ public DeleteRequest process(DeleteRequest input)
         return input;
     }
 
+    @Override
+    public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+        if (Requests.isLocal(input.getRequest())) {
+            for (Metacard metacard : input.getCreatedMetacards()) {
+                if (metacard.getTags()
+                        .contains(RegistryConstants.REGISTRY_TAG)) {
+                    registryIds.add(getRegistryId(metacard));
+                }
+            }
+        }
+        return input;
+    }
+
+    @Override
+    public UpdateResponse process(UpdateResponse input) throws PluginExecutionException {
+        return input;
+    }
+
+    @Override
+    public DeleteResponse process(DeleteResponse input) throws PluginExecutionException {
+        if (Requests.isLocal(input.getRequest())) {
+            for (Metacard metacard : input.getDeletedMetacards()) {
+                if (metacard.getTags()
+                        .contains(RegistryConstants.REGISTRY_TAG)) {
+                    registryIds.remove(getRegistryId(metacard));
+                }
+            }
+        }
+        return input;
+    }
+
     private void setMetacardExtID(Metacard metacard) throws StopProcessingException {
 
         boolean extOriginFound = false;
         String metacardID = metacard.getId();
         String metadata = metacard.getMetadata();
-        String registryID = metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
-                .getValue()
-                .toString();
+        String registryID = getRegistryId(metacard);
 
         InputStream inputStream = new ByteArrayInputStream(metadata.getBytes(Charsets.UTF_8));
-        FileBackedOutputStream outputStream = new FileBackedOutputStream(1000000);
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
 
         try {
             JAXBElement<RegistryObjectType> registryObjectTypeJAXBElement = parser.unmarshal(
```

Comment:
We are doing this marshalling and replacing the ID in another part of the code - we should only have this logic done once.  I think this could be done much simpler with some XPath replacements instead of marshalling & unmarshalling with JAXB which can be expensive.